packages/components/app/styles/components/application-state.scss

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// APPLICATION-STATE COMPONENT
+//
+$hds-application-state-padding: 12px 0;
+
+.hds-application-state {
+  align-self: center;
+  width: 20rem;
+  max-width: 100%;
+  margin: 0 auto;
+}
+
+.hds-application-state__body,
+.hds-application-state__footer {
+  display: flex;
+}
+
+.hds-application-state__header {
+  display: grid;
+  grid-row-gap: 0;
+  grid-column-gap: 0;
+  color: var(--token-color-foreground-primary);
+
+  .hds-application-state--error & {
+    grid-template-rows: repeat(2, auto);
+    grid-template-columns: 28px auto;
+    align-items: stretch;
+
+    .hds-application-state__header-icon {
+      grid-area: 1 / 1 / 3 / 2;
+      padding-top: 4px;
+    }
+
+    .hds-application-state__header-title {
+      grid-area: 1 / 2 / 2 / 3;
+    }
+
+    .hds-application-state__header-errorCode {
+      grid-area: 2 / 2 / 3 / 3;
+    }
+  }
+}
+
+.hds-application-state__body {
+  padding: $hds-application-state-padding;
+  color: var(--token-color-foreground-faint);
+}
+
+.hds-application-state__footer {
+  display: flex;
+  justify-content: space-between;
+
+  .hds-application-state--has-divider & {
+    padding: $hds-application-state-padding;
```

Comment:
I'll check again, but here's what I'm seeing: 

![CleanShot 2023-03-29 at 16 15 58](https://user-images.githubusercontent.com/4587451/228669597-c91ff0b0-097b-404f-a472-382d073777d7.png)

![CleanShot 2023-03-29 at 16 13 55](https://user-images.githubusercontent.com/4587451/228669244-6d3eb091-3ce6-4254-a182-f89270822c1d.png)

Reply:
I should've been a bit more clear, I was referring to the footer padding when there is a divider

<img width="541" alt="Screenshot 2023-03-30 at 09 57 10" src="https://user-images.githubusercontent.com/788096/228784686-1a87417a-7873-4fa8-9914-f313892f5213.png">

Reply:
I've asked @andgen404 to review the design details to see if that's what was intended (since the component is otherwise symmetrical in the vertical spacing)

Reply:
@MelSumner, The spacing in the footer has been updated in Figma to 12px instead of 16px when a divider is used.